metaflow/plugins/metadata/service.py

```diff
@@ -314,30 +375,45 @@ def create_object():
             return create_object()
 
         try:
-            return self._request(self._monitor, obj_path)
+            return self._request(self._monitor, obj_path, "GET")
         except ServiceException as ex:
             if ex.http_code == 404:
                 return create_object()
             else:
                 raise
 
+    # TODO _request() needs a more deliberate refactor at some point, it looks quite overgrown.
     @classmethod
-    def _request(cls, monitor, path, data=None, retry_409_path=None):
+    def _request(
+        cls,
+        monitor,
+        path,
+        method,
+        data=None,
+        retry_409_path=None,
+        return_raw_resp=False,
+    ):
         if cls.INFO is None:
             raise MetaflowException(
                 "Missing Metaflow Service URL. "
                 "Specify with METAFLOW_SERVICE_URL environment variable"
             )
+        supported_methods = ("GET", "PATCH", "POST")
+        if method not in supported_methods:
+            raise MetaflowException(
+                "Only these methods are supported: %s, but got %s"
+                % (supported_methods, method)
+            )
         url = os.path.join(cls.INFO, path.lstrip("/"))
         for i in range(METADATA_SERVICE_NUM_RETRIES):
             try:
-                if data is None:
+                if method == "GET":
```

Comment:
This is a drive-by.  Simply using the presence of `data` to determine the HTTP method was tenuous before and does not work at all, now that we want to use PATCH.